DICOMPlugins/DICOMTID1500Plugin.py

```diff
@@ -332,6 +360,80 @@ def enumerateDuplicateNames(self, items):
       items[idx]["name"] = item
     return items
 
+  def isConcept(self, item, coding):
+    code = item.ConceptNameCodeSequence[0]
+    return code.CodingSchemeDesignator == self.codings[coding]["scheme"] and code.CodeValue == self.codings[coding]["value"]
+
+  def loadAdditionalMeasurements(self, srUID):
+    """
+    Loads length measements as annotation rulers
+    TODO: need to generalize to other report contents
+    """
+
+    srFilePath = slicer.dicomDatabase.fileForInstance(srUID)
+    sr = dicom.read_file(srFilePath)
+
+    if not self.isConcept(sr, "imagingMeasurementReport"):
+      return sr
+
+    contents = {}
+    measurements = []
+    contents['measurements'] = measurements
+    for item in sr.ContentSequence:
+      if self.isConcept(item, "personObserver"):
+        contents['personObserver'] = item.PersonName
+      if self.isConcept(item, "imagingMeasuremnts"):
+        for contentItem in item.ContentSequence:
+          if self.isConcept(contentItem, "measurementGroup"):
+            measurement = {}
+            for measurementItem in contentItem.ContentSequence:
+              if self.isConcept(measurementItem, "trackingIdentifier"):
+                measurement['trackingIdentifier'] = measurementItem.TextValue
+              if self.isConcept(measurementItem, "trackingUniqueIdentifier"):
+                measurement['trackingUniqueIdentifier'] = measurementItem.UID
+              if self.isConcept(measurementItem, "findingSite"):
+                measurement['findingSite'] = measurementItem.ConceptCodeSequence[0].CodeMeaning
+              if self.isConcept(measurementItem, "length"):
+                for lengthItem in measurementItem.ContentSequence:
+                  measurement['polyline'] = lengthItem.GraphicData
+                  for selectionItem in lengthItem.ContentSequence:
+                    if selectionItem.RelationshipType == "SELECTED FROM":
```

Comment:
I am a bit confused as to how this parses the content - should it be checking for the presence of SCOORD in the first place?

```text
  <contains CONTAINER:(,,"Imaging Measurements")=SEPARATE>
    <contains CONTAINER:(,,"Measurement Group")=SEPARATE>
      <has obs context TEXT:(,,"Tracking Identifier")="5b6eb4301d3175942d29985a3d0...">
      <has obs context UIDREF:(,,"Tracking Unique Identifier")="1.3.6.1.4.1.5962.1.1.0.0.0.1535644357.22655.1">
      <has concept mod CODE:(,,"Finding Site")=(T-71000,SRT,"Kidney")>
      <contains NUM:(,,"Length")="66.43856134" (mm,UCUM,"mm")>
        <inferred from SCOORD:=(POLYLINE,172.835/270.0641,...)>
          <selected from IMAGE:=(CT image,)>
```

Reply:
Let's generate (or collect) some examples of the types of documents we want to support and then we can generalize the code.